daemon/log.c

```diff
@@ -299,11 +263,7 @@ void log_message(MTC_S32 priority, PMTC_S8 fmt, ...)
         va_end(ap);
     }
 
-#if USE_SEMAPHORE
-    sem_post(semaphore);
-#else
-    pthread_spin_unlock(&lock);
-#endif
+    pthread_rwlock_unlock(&lock);
```

Comment:
We should unlock once we've done the logfile bit, before looking at syslog, as we don't need to hold the lock while syslogging